src/extension/experimentService/experimentService.ts

```diff
@@ -0,0 +1,130 @@
+        this.downloadConfigProm = null;
+    }
+
+    public async initialize(): Promise<void> {
+        this.downloadConfigProm = this.downloadExperimentsConfig();
+    }
+
+    public async runExperiments(): Promise<void> {
+        if (!this.downloadedExperimentsConfig) {
+            if (!this.downloadConfigProm) {
+                throw new Error("Experiment Service is not initialized");
+            }
+            try {
+                this.downloadedExperimentsConfig = await this.downloadConfigProm;
+            } catch (err) {
+                this.downloadConfigProm = this.downloadExperimentsConfig();
+                throw new Error("Failed to download experiments config");
+            }
+        }
+
+        let experimentResults: Array<ExperimentResult> = await Promise.all(this.downloadedExperimentsConfig
+            .filter(expConfig => expConfig.enabled)
+            .map(async (expConfig) => await this.executeExperiment(expConfig))
+        );
+
+        this.sendExperimentTelemetry(experimentResults);
+    }
+
+    private async executeExperiment(expConfig: ExperimentConfig): Promise<ExperimentResult> {
+        let curExperimentParameters = this.config.get(expConfig.experimentName);
+        let expResult: ExperimentResult;
+
+        try {
+            switch (expConfig.experimentName) {
+                case this.availableExperiments.RNT_PREVIEW_PROMPT:
+                    let previewPromptExp = new PreviewVersionPromotion();
+                    expResult = await previewPromptExp.run(expConfig, curExperimentParameters);
+                    break;
+                default:
+                    throw new Error("Cannot run the experiment. There is no such experiment.");
+            }
+        } catch (err) {
+            return {
+                resultStatus: ExperimentStatuses.FAILURE,
+                updatedExperimentParameters: expConfig,
+            };
+        }
+
+        this.config.set(expConfig.experimentName, expResult.updatedExperimentParameters);
+        return expResult;
+    }
+
+    private downloadExperimentsConfig(): Promise<ExperimentConfig[]> {
+        return new Promise<ExperimentConfig[]>((resolve, reject) => {
+            https.get(this.endpointURL, (response) => {
+                let data = "";
+                response.setEncoding("utf8");
+                response.on("data", (chunk: string) => (data += chunk));
+                response.on("end", () => {
+                    try {
+                        resolve(JSON.parse(data));
+                    } catch (err) {
+                        reject(err);
+                    }
+                });
+                response.on("error", reject);
+            }).on("error", reject);
+        });
+    }
+
+    private sendExperimentTelemetry(experimentsResults: ExperimentResult[]): void {
+        const telemetryProps = experimentsResults.reduce((tProps, expResult) => {
+            return Object.assign(
```

Comment:
Could you please add description for the new telemetry event in the PR description?